File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*; 

class Solution {
    
    public String solution(int[] numbers) {
        
        String num[] = new String[numbers.length];
        
        for(int i=0; i<numbers.length; i++){
            num[i] = numbers[i]+"";
        }
        
        Arrays.sort(num, (o1, o2) -> (o2 + o1).compareTo(o1 + o2));
        String answer = "";
        
        if (num[0].equals("0")) {
            return "0";
        }
        
        for(int i=0; i<num.length; i++){
            answer = answer + num[i];
        }
        
        return answer;
    }
    
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.js
function solution(numbers) {
    
    const num = Array.from({length : numbers.length}).fill("");
    
    for(let i=0; i< num.length; i++){
        num[i] = numbers[i]+"";
    }
    
    num.sort((o1, o2) => (o2+o1).localeCompare(o1+o2));
    
    if(num[0] == "0") return "0";
    
    var answer = num.join("");

    return answer;
}